Slack.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Slack;
+
+use Piwik\Log\LoggerInterface;
+use Piwik\Piwik;
+use Piwik\Plugin;
+use Piwik\Plugins\ScheduledReports\ScheduledReports;
+use Piwik\View;
+use Piwik\Container\StaticContainer;
+use Piwik\ReportRenderer;
+use Piwik\Plugins\ScheduledReports\API as APIScheduledReports;
+
+class Slack extends Plugin
+{
+    public const SLACK_CHANNEL_ID_PARAMETER = 'slackChannelID';
+    public const SLACK_TYPE = 'slack';
+    public const EVOLUTION_GRAPH_PARAMETER = 'evolutionGraph';
+    public const DISPLAY_FORMAT_PARAMETER = 'displayFormat';
+    public const DISPLAY_FORMAT_GRAPHS_ONLY = 2; // Display Graphs Only for all reports
+    public const DISPLAY_FORMAT_TABLES_AND_GRAPHS = 3; // Display Tables and Graphs for all reports
+    public const DISPLAY_FORMAT_TABLES_ONLY = 4; // Display only tables for all reports
+
+    private static $availableParameters = array(
+        self::SLACK_CHANNEL_ID_PARAMETER    => true,
```

Comment:
Why is this set to 'true'? Shouldn't it be a string value?

Nitpick: The spacing is odd. I'm surprised that PHPCS didn't pick it up.
```suggestion
        self::SLACK_CHANNEL_ID_PARAMETER => true,
```

Reply:
This indicate that the parameter is a compulsory parameter, hence required true, same as what we have for [MobileMessaging](https://github.com/matomo-org/matomo/blob/5.x-dev/plugins/MobileMessaging/MobileMessaging.php#L40) plugin.